mypy/server/update.py

```diff
@@ -1108,3 +1109,51 @@ def target_from_node(module: str,
             return '%s.%s' % (node.info.fullname, node.name)
         else:
             return '%s.%s' % (module, node.name)
+
+
+def refresh_suppressed_submodules(
+        module: str,
+        path: Optional[str],
+        deps: Dict[str, Set[str]],
+        graph: Graph,
+        fscache: FileSystemCache) -> None:
+    """Look for submodules that are now suppressed in target package.
+
+    If a submodule a.b gets added, we need to mark it as suppressed
+    in modules that contain "from a import b". Previously we assumed
+    that 'a.b' is not a module but a regular name.
+
+    This is only relevant when following imports normally.
+
+    Args:
+        module: target package in which to look for submodules
+        path: path of the module
+    """
+    # TODO: Windows paths
+    if path is None or not path.endswith(os.sep + '__init__.py'):
+        # Only packages have submodules.
+        return
+    # Find and submodules present in the directory.
+    pkgdir = os.path.dirname(path)
+    for fnam in fscache.listdir(pkgdir):
+        if (not fnam.endswith(('.py', '.pyi'))
+                or fnam.startswith("__init__.")
+                or fnam.count('.') != 1):
+            continue
+        shortname = fnam.split('.')[0]
+        submodule = module + '.' + shortname
+        trigger = make_trigger(submodule)
+        if trigger in deps:
+            for dep in deps[trigger]:
+                # TODO: <...> deps, imports in functions, etc.
```

Comment:
Do we think this approach will scale to handling this stuff?

Reply:
I hope so, with some further optimizations perhaps. Following imports might not be very performant for code bases with more than (say) 500 kLOC, but even if that's the case, it would still probably be fine for a significant majority of users. Users with very large codebases can continue to disable following imports, and we just need to document this clearly.

 It sure would be better to have this perform well with large code bases, but I'm not sure how much extra work that would take.